File: problem_013/13_test.go
package main

import (
	"testing"
)

func TestSumDigitsOfANumber(t *testing.T) {
	expected := uint(9)
	got := SumDigitsOf(27)
	if expected != got {
		t.Errorf("%d != %d", expected, got)
	}
}

func TestSliceDigitsOfANumber(t *testing.T) {
	expected := "1234"
	got := SliceDigits(123456789, 4)
	if expected != got {
		t.Errorf("%s != %s", expected, got)
	}
}

func TestReduce(t *testing.T) {
	input := []uint{1, 2, 3, 4}
	expected := uint(10)
	got := Reduce(func(x, y uint) uint { return x + y }, input)
	if expected != got {
		t.Errorf("%d != %d", expected, got)
	}
}

File: problem_013/python/13_test.py
import unittest
from problem13 import slice_digits, sum_digits_of

class Problem13TestCase(unittest.TestCase):

    def test_can_sum_digits_of_a_number(self):
        expected = 9
        got = sum_digits_of(27)
        self.assertEquals(expected, got)

    def test_can_slice_digits_of_a_number(self):
        expected = "1234"
        got = slice_digits(123456789, 4)

unittest.main()
